packages/next/client/link.tsx

```diff
@@ -201,55 +222,77 @@ function Link(props: React.PropsWithChildren<LinkProps>) {
     // This hook is in a conditional but that is ok because `process.env.NODE_ENV` never changes
     // eslint-disable-next-line react-hooks/rules-of-hooks
     const hasWarned = React.useRef(false)
-    if (props.prefetch && !hasWarned.current) {
+    if (oldBehavior && props.prefetch && !hasWarned.current) {
```

Comment:
Should we still show this warning as `prefetch` shouldn't need to be manually set with the new behavior either?